src/Reflow/Parser.hs

```diff
@@ -31,7 +31,10 @@ header = do
     return $ Header $ n <> v
 
 headerName :: Parser Text
-headerName = pack <$> (many1 $ noneOf ":\n")
+headerName = do
+    n <- manyTill (noneOf "\n") (lookAhead $ string ":")
+    s <- string ":"
+    return $ pack $ n <> s
```

Comment:
Ah, yes, _that's_ what this should look like. I could _not_ figure out how this was supposed to look with applicative style. If I consume the `:`, this could be even nicer:

```haskell
pack . mappend <$> manyTill (noneOf "\n") (string ":") <*> pure ":"
```